src/services/storage/models/offer.model.ts

```diff
@@ -50,33 +50,37 @@ export default class Offer extends Model {
   get availableCapacity (): BigNumber {
     return this.totalCapacity.minus(this.utilizedCapacity)
   }
+
+  @Column(DataType.VIRTUAL)
+  get acceptedCurrencies (): Array<SupportedTokens> {
+    return Array.from(
+      new Set(
+        (this.plans || []).map(plan => plan.rateId)
+      )
+    )
+  }
 }
 
 /**
  * This function generate nested query for aggregating total stakes for offer in specific currency
  * @param sequelize
  * @param currency
  */
-export async function getStakesAggregateQuery (sequelize: Sequelize, currency: 'usd' | 'eur' | 'btc' = 'usd') {
-  if (!config.get('storage.tokens')) {
-    throw new Error('"storage.tokens" not exist in config')
-  }
-
-  const supportedTokens = Object.entries(config.get('storage.tokens'))
-  const rates = await Rate.findAll()
-  return literal(`(
-    SELECT SUM(
-      case
-        ${supportedTokens.reduce(
-          (acc, [tokenAddress, tokenSymbol]) => {
-            const rate: number = rates.find(r => r.token === tokenSymbol)?.[currency] || 0
-            return `${acc} \n when token = ${sequelize.escape(tokenAddress)} then cast(total as integer) * ${sequelize.escape(rate)}`
-          },
-          ''
-        )}
-        else 0
-      end
-    ) from storage_stakes where account = provider)
+export function getStakesAggregateQuery (
+  sequelize: Sequelize,
+  currency: 'usd' | 'eur' | 'btc' = 'usd'
+): Literal {
+  return literal(`
+  (
+    SELECT
+      SUM(total * coalesce("rates"."${currency}", 0))
```

Comment:
Yeah, this query looks so better! Good job!

Reply:
Yea, thanks for suggesting